ChatSchema.md

```diff
@@ -0,0 +1,105 @@
+# Chat Server-side Schema
+
+## Entity Design
+
+### User
+
+User entity includes:
+
+- user_id: identify a user. (Pk)
+- name: shown name.
+- password: login password (optional).
+- create_time: create time.
+- update_time: update time.
+- visual_id: linked visual charactor.
+- data: map(str->str) containing additional information like last login.
+
+User relationship entity includes:
+
+- user_id: first user_id.
+- firend_id: second user_id.
+- create_time: create time.
+- update_time: update time.
+- type: `friend` or `block` or `NULL`
+
+Relationship is one-way, like telegram.
+
+### Chat
+
+Chat entity includes:
+
+- chat_id: identify a chat. (Pk)
+- name: chat name that can be searched.
+- create_time: create time.
+- update_time: update time.
+- data: map(str->str) containing additional information like group icon.
+- is_p2p: not group chat.
+
+Chat Enrollment entity includes:
+
+- chat_id: (1st of Pk)
+- user_id: (2nd of Pk)
```

Comment:
If use joins the chat room with chat id, then seems we don't need to maintain this relationship.